Assets/Scripts/Interactables/SodaCan.cs

```diff
@@ -81,12 +87,16 @@ private void StartFlying()
         state = SodaCanState.Flying;
         sprayEffect.SendEvent(VisualEffectAsset.PlayEventID);
         sprayEffect.SetBool("IsSpraying", true);
+        audioSource.loop = true;
+        sprayNoise.PlayExclusively(audioSource);
     }
 
     private void StopFlying()
     {
         state = SodaCanState.Emptied;
         sprayEffect.SetBool("IsSpraying", false);
+        audioSource.loop = false;
+        audioSource.Stop();
```

Comment:
Unsure if you need both calls here

Reply:
These lines are here because both the explosion and the spray noise use the same audio source, so I need to ensure they dont overlap

Reply:
Aha